src/ceph-daemon/ceph-daemon

```diff
@@ -1616,6 +1616,83 @@ def command_rm_cluster():
 
 ##################################
 
+def check_time_sync():
+    units = ['chronyd.service', 'systemd-timesyncd.service', 'ntpd.service']
+    for u in units:
+        (enabled, state) = check_unit(u)
+        if enabled and state == 'running':
+            logger.info('Time sync unit %s is enabled and running' % u)
+            return True
+    logger.warning('No time sync service is running; checked for %s' % units)
+    return False
+
+def command_check_host():
+    # caller already checked for docker/podman
+    logger.info('podman|docker (%s) is present' % container_path)
+
+    if not find_program('systemctl'):
+        raise RuntimeError('unable to location systemctl')
+    logger.info('systemctl is present')
+
+    if not find_program('lvcreate'):
+        raise RuntimeError('LVM does not appear to be installed')
+    logger.info('LVM2 is present')
+
+    # check for configured+running chronyd or ntp
+    if not check_time_sync():
+        raise RuntimeError('No time synchronization is active (checked all of %s)' %
+                           units)
+
+    logger.info('Host looks OK')
+
+def command_prepare_host():
+    # if there is no systemd we're SOL
+    if not find_program('systemctl'):
+        raise RuntimeError('unable to location systemctl')
+
+    if not container_path:
+        logger.info('Trying to install podman or docker...')
+        out, err, ret = call(['dnf', 'install', '-y', 'podman'])
```

Comment:
I kinda have my problems with installing stuff from within ceph-daemon. This will be a point where people will start to talk about feature creep.

Reply:
They will definitely talk about it :).  IMO the question is: does this make sense for a good user experience with mgr/ssh.  If so, then this is a small price to pay.

Reply:
How do you deploy kernel updates? If you have a cluster large enough to require an automated solution for this, then you probably already have a solution to install software. I'd think it therefore only applies to smaller clusters, where installing CD by hand is not a requirement as a MVP

Reply:
It's not a small price, since it also opens up the need to handle various failures (as this rather interesting `if err:` block that follows indicates, at the very least we ought to detect the flavor we're running on and then do the appropriate thing, which is going to bake in assumptions, instead of that copy-paste-repetition approach ;-) ).